vunit/vivado/vivado.py

```diff
@@ -119,12 +133,22 @@ def run_vivado(tcl_file_name, tcl_args=None, cwd=None, vivado_path=None):
     """
     Run tcl script in Vivado in batch mode.
 
-    Note: the shell=True is important in windows where Vivado is just a bat file.
+    :param tcl_file_name: Path to tcl file
+    :param tcl_args: tcl arguments passed to Vivado via the ``-tclargs`` switch
+    :param cwd: Passed as ``cwd`` to :func:`subprocess.check_call`
+    :param vivado_path: Path to Vivado install directory. If ``None``, the
+        environment variable ``VUNIT_VIVADO_PATH`` is used if set. Otherwise,
+        rely on that ``vivado`` is in the path.
     """
-    vivado = "vivado" if vivado_path is None else str(Path(vivado_path).resolve() / "bin" / "vivado")
-    cmd = f"{vivado} -nojournal -nolog -notrace -mode batch -source {str(Path(tcl_file_name).resolve())}"
+    vivado = (
+        str(Path(vivado_path).resolve() / "bin" / "vivado")
+        if vivado_path is not None
+        else environ.get("VUNIT_VIVADO_PATH", "vivado")
+    )
+    cmd = f'"{vivado}" -nojournal -nolog -notrace -mode batch -source "{Path(tcl_file_name).resolve()}"'
     if tcl_args is not None:
-        cmd += " -tclargs " + " ".join([str(val) for val in tcl_args])
+        cmd += " -tclargs " + " ".join([f'"{val}"' for val in tcl_args])
 
     print(cmd)
+    # shell=True is important in Windows where Vivado is just a bat file.
     check_call(cmd, cwd=cwd, shell=True)
```

Comment:
Probably not to be done in this PR, but `check_call`'s `args` parameter can be either a list of a string: https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.args.
I feel it would be cleaner if we managed `cmd` as a list, so we could append `tcl_args` without explicitly using `" ".join`. See https://docs.python.org/3/library/subprocess.html#converting-an-argument-sequence-to-a-string-on-windows.